### PR TITLE
Fix `prefer-object-spread` in `/scripts`

### PR DESCRIPTION
**Pull request by fisker**
- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: scripts/generate-schema.js [hunk 1]
```diff
@@ -80,17 +78,15 @@ function generateSchema(options) {
...
+  return {
+    description: option.description,
+    default: option.default,
+    ...(option.array ? wrapWithArraySchema : identity)(
       option.type === "choice"
         ? { oneOf: option.choices.map(choiceToSchema) }
         : { type: optionTypeToSchemaType(option.type) }
     )
-  );
+  };
```
Comment (fisker) [inline scripts/generate-schema.js]: I think no tests cover this part, need review.

File: scripts/generate-schema.js [hunk 2]
```diff
@@ -21,10 +21,8 @@ function generateSchema(options) {
     definitions: {
       optionsDefinition: {
         type: "object",
-        properties: options.reduce(
-          (props, option) =>
-            Object.assign(props, { [option.name]: optionToSchema(option) }),
-          {}
+        properties: Object.assign(
+          ...options.map(option => ({ [option.name]: optionToSchema(option) }))
         )
```
Comment (fisker) [inline scripts/generate-schema.js]: I think no tests cover this part, need review.

File: scripts/generate-schema.js (hunk 1 above)
Comment (fisker) [inline scripts/generate-schema.js]: And this

File: scripts/generate-schema.js (hunk 2 above)
Comment (thorn0) [reply scripts/generate-schema.js]: `tests_integration\__tests__\schema.js` seems to cover it.

**Pull request merged**